test/CMakeLists.txt

```diff
@@ -110,6 +110,10 @@ macro(onedpl_add_test test_source_file switch_off_checked_iterators)
 
     set_tests_properties(${_test_name} PROPERTIES SKIP_RETURN_CODE 77)
 
+    if ("${_test_name}" STREQUAL selected_different_device.pass AND _ONEDPL_PSTL_OFFLOAD MATCHES "(cpu|gpu)")
+        set(DEVICE_SELECTION_LINE "${DEVICE_SELECTION_VARIABLE}=${ONEDPL_DEVICE_BACKEND}:${_ONEDPL_PSTL_OFFLOAD}")
+    endif()
```

Comment:
I don't completely understand what these lines are intended for. Could you please explain?

Reply:
I'll add a comment, but generally there are 2 options.

1. Set ONEAPI_DEVICE_SELECTOR always according to PSTL offload policy.
2. Set it only when required.

I'm slightly for (2), as it more aligned with possible working mode. It should be noted that (1) leads to more clear cmake code, but this is not a goal.